### PR TITLE
Support for SSTL15 IO standard.

### PR DESCRIPTION
**Pull request by mkurc-ant**
This PR adds support for SSTL15 using fasm features as defined in https://github.com/SymbiFlow/prjxray/pull/1324. The `xc/xc7/tests/iosettings` test was upgraded as well as `fasm2bels` to handle that IO standard.

~This PR requires database with the new features, I expect CI to fail now.~

### CODE REVIEW

**Comment by mkurc-ant** [comment]
@litghost No other CI fails apart from https://github.com/SymbiFlow/symbiflow-arch-defs/issues/1400

File: xc/xc7/tests/iosettings/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,10 @@
-#add_custom_target(all_xc7_iosettings_diff_fasm)
+# These targets make Kokoro complain that the Ninja version is too low.
+#add_custom_target(all_xc7_iosettings_generate)
+#add_custom_target(all_xc7_iosettings_bin)
+#add_custom_target(all_xc7_iosettings_bit_v)
+#add_custom_target(all_xc7_iosettings_vivado_diff_fasm)
 
-function(xc7_iosettings_design MODE BOARD IOSTANDARD DRIVES SLEWS TERMS)
+function(xc7_iosettings_design MODE BOARD IOSTANDARD DRIVES SLEWS TERMS VREF)
```
Comment (litghost) [inline xc/xc7/tests/iosettings/CMakeLists.txt:7]: Please convert this function to use named arguments calling convention, using cmake_parse_arguments.  It is hard to parse a function with 7 arguments.

Comment (mkurc-ant) [reply xc/xc7/tests/iosettings/CMakeLists.txt]: Ok, I've cleaned that up.

File: xc/xc7/tests/iosettings/CMakeLists.txt [hunk 2]
```diff
@@ -58,38 +98,110 @@ function(xc7_iosettings_design MODE BOARD IOSTANDARD DRIVES SLEWS TERMS)
...
 
-# Currently only one differential setting is supported
-xc7_iosettings_design("diff_output" ${BOARD} DIFF_SSTL135 "0" "SLOW FAST" "NONE")
+# Differential. Currently only differential output (tristated) is supported.
+foreach(IOSTANDARD DIFF_SSTL135 DIFF_SSTL15)
+    xc7_iosettings_design(
+        MODE "diff_output"
+        BOARD ${BOARD}
+        IOSTANDARD ${IOSTANDARD}
+        DRIVES "0"
```
Comment (litghost) [inline xc/xc7/tests/iosettings/CMakeLists.txt:202]: Rather than DRIVES "0", maybe DRIVES FIXED?

**Comment by litghost** [comment]
CI is red across the board?

File: xc/xc7/tests/iosettings/CMakeLists.txt (hunk 2 above)
Comment (mkurc-ant) [reply xc/xc7/tests/iosettings/CMakeLists.txt]: Ok, Now you simply don't give DRIVE argument for IOSTANDARDs that do not require it.

**Comment by mkurc-ant** [comment]
@litghost Travis CI failed on Python lint showing errors on files untouched by this PR. Apparently conda packages `flake8` and `pyflakes` changed in the conda repo and we do not enforce their versions.

Previously was:
```
flake8                    3.7.9            py37hc8dfbb8_1    conda-forge
pyflakes                  2.1.1                    py37_0
```
And now is:
```
flake8                    3.8.1              pyh9f0ad1d_0    conda-forge
pyflakes                  2.2.0              pyh9f0ad1d_0    conda-forge
```

Thoughts?

**Comment by acomodi** [comment]
@mkurc I have fixed those python linter issues here: https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1484/commits/740ffdb58bc45cbf67146f0542f80188f5cca5d0.

I think it is ok to keep the latest linter version, as it keeps the scripts standardized

**Comment by mkurc** [comment]
@acomodi this is not the mkurc you're looking for

**Comment by acomodi** [comment]
@mkurc Oh, right, my bad. Sorry about that :)

**Comment by acomodi** [comment]
@mkurc-ant ^

**Comment by mkurc-ant** [comment]
Thanks @acomodi, I've cherry-picked that commit.

File: utils/lib/pb_type_xml.py [hunk 3]
```diff
@@ -100,6 +100,19 @@ def add_switchblock_locations(xml):
     })
 
 
+def get_site_pin_wire(equivalent_site_pin, site_pins):
+    no_prefix_equivalent_site_pin_wire = equivalent_site_pin.wire.split('_'
```
Comment (litghost) [inline utils/lib/pb_type_xml.py:104]: This function needs a docstring.

File: xc/xc7/tests/iosettings/CMakeLists.txt [hunk 4]
```diff
@@ -1,50 +1,100 @@
-#add_custom_target(all_xc7_iosettings_diff_fasm)
+# These targets make Kokoro complain that the Ninja version is too low.
+#add_custom_target(all_xc7_iosettings_generate)
+#add_custom_target(all_xc7_iosettings_bin)
+#add_custom_target(all_xc7_iosettings_bit_v)
+#add_custom_target(all_xc7_iosettings_vivado_diff_fasm)
 
-function(xc7_iosettings_design MODE BOARD IOSTANDARD DRIVES SLEWS TERMS)
+function(xc7_iosettings_design)
```
Comment (litghost) [inline xc/xc7/tests/iosettings/CMakeLists.txt:7]: This function needs a documentation string.

File: utils/lib/pb_type_xml.py (hunk 3 above)
Comment (mkurc-ant) [reply utils/lib/pb_type_xml.py:104]: @acomodi Can you help me with the docstring there?

File: xc/xc7/tests/iosettings/CMakeLists.txt (hunk 4 above)
Comment (mkurc-ant) [reply xc/xc7/tests/iosettings/CMakeLists.txt:7]: Added docstring.

File: utils/lib/pb_type_xml.py (hunk 3 above)
Comment (acomodi) [reply utils/lib/pb_type_xml.py:104]: @mkurc-ant This change went in the commit with the flake8 linter fixes, the docstrings can be found here: https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1484/commits/2e4d5a332b208a55510bdcac8d0b5d7e4aabf439 and here is also a fix for the assertion: https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1484/commits/357add1d840d92666b8a373c3856cc03bf03efc2

Comment (mkurc-ant) [reply utils/lib/pb_type_xml.py:104]: Thx

**Comment by mkurc-ant** [comment]
The `VtR + SymbiFlow Architecture Definitions (Presubmit)` test uses wrong version of VPR - `7d6424bb0` instead of `0ece2d2c1` (no `--quick_check_route`).

The `Xilinx Series 7 - Vendor Tool Tests (Presubmit)` fails (as usually) due to incorrect VCC routing.
